tiny_dnn/layers/deconvolutional_layer.h

```diff
@@ -198,7 +214,12 @@ class deconvolutional_layer : public layer {
                         size_t w_stride        = 1,
                         size_t h_stride        = 1,
                         backend_t backend_type = core::default_engine())
-    : layer(std_input_order(has_bias), {vector_type::data}) {
+    : layer({vector_type::data}, {vector_type::data}) {
+    layer::add_parameter(out_channels, in_channels, window_height, window_width,
+                         parameter_type::weight);
```

Comment:
Yes, since I am doing parameter integration layer by layer, I am duplicating older methods and data structures with similar names. Once all layers will have it integrate, a huge chunk of code will vanish / get merged with its duplicate counterpart.. This is just a temporary transient phase and not a final API :wink: Please do let me know if you vision some idea about how final Parameter API should look!